staging/src/k8s.io/client-go/tools/record/util/util.go

```diff
@@ -38,3 +41,27 @@ func IsKeyNotFoundError(err error) bool {
 
 	return statusErr != nil && statusErr.Status().Code == http.StatusNotFound
 }
+
+// GenerateEventName generates a valid Event name from the referenced name and the passed UNIX timestamp.
+// The referenced Object name may not be a valid name for Events and cause the Event to fail
+// to be created, so we need to generate a new one in that case.
+// Ref: https://issues.k8s.io/127594
+func GenerateEventName(refName string, unixNano int64) string {
+	name := fmt.Sprintf("%s.%x", refName, unixNano)
+	if errs := apimachineryvalidation.NameIsDNSSubdomain(name, false); len(errs) > 0 {
+		// Name must be a lowercase RFC 1123 subdomain, that  must consist of lower case alphanumeric
+		// characters, '-' or '.' and max length of 253 characters.
+		// Using a length of 63 gives us 36^63 possible combinations, that adding the timestamp as suffix
+		// reduces any chance of collision.
+		length := 63
+		// use the timestamp + the refName sum of ascii value as seed
+		// the refName is used to differentiate between multiple events in the same timestamp
+		seed := unixNano
+		for _, char := range refName {
+			seed += int64(char) // char is a rune, which is an int32
+		}
+		apimachineryrand.Seed(seed)
+		name = fmt.Sprintf("%s.%x", apimachineryrand.String(length), unixNano)
```

Comment:
not sure re-seeding and using a rand string generator that requires a global lock is a great idea... if the format produced by uuid.New() is valid for this, maybe just use that?

Reply:
it was just for making the unit test deterministic, the number of times this is going to be called is really small

Reply:
hrmm... I don't love doing things for unit test determinism in a non test file

Reply:
uuid is fine, will switch to it then